src/constraint/uniform/SE2BoxConstraint.cpp

```diff
@@ -0,0 +1,213 @@
+#include <stdexcept>
+#include <aikido/constraint/uniform/SE2BoxConstraint.hpp>
+#include <aikido/constraint/uniform/SO2UniformSampler.hpp>
+#include <aikido/statespace/SO2.hpp>
+
+namespace aikido {
+namespace constraint {
+
+using constraint::ConstraintType;
+
+//=============================================================================
+class SE2BoxConstraintSampleGenerator
+  : public constraint::SampleGenerator
+{
+public:
+  statespace::StateSpacePtr getStateSpace() const override;
+
+  bool sample(statespace::StateSpace::State* _state) override;
+
+  int getNumSamples() const override;
+
+  bool canSample() const override;
+
+private:
+  SE2BoxConstraintSampleGenerator(
+    std::shared_ptr<statespace::SE2> _space,
+    std::unique_ptr<util::RNG> _rng,
+    const Eigen::Vector3d& _lowerLimits,
+    const Eigen::Vector3d& _upperLimits);
+
+  const size_t mDimension;
+  std::shared_ptr<statespace::SE2> mSpace;
+  std::unique_ptr<util::RNG> mRng;
+  std::vector<std::uniform_real_distribution<double>> mDistributions;
+  friend class SE2BoxConstraint;
+};
+
+//=============================================================================
+SE2BoxConstraintSampleGenerator::SE2BoxConstraintSampleGenerator(
+      std::shared_ptr<statespace::SE2> _space,
+      std::unique_ptr<util::RNG> _rng,
+      const Eigen::Vector3d& _lowerLimits,
+      const Eigen::Vector3d& _upperLimits)
+  : mDimension(3)
+  , mSpace(std::move(_space))
+  , mRng(std::move(_rng))
+{
+  mDistributions.reserve(mDimension);
+
+  for (size_t i = 0; i < mDimension; ++i)
+    mDistributions.emplace_back(_lowerLimits[i], _upperLimits[i]);
+}
+
+//=============================================================================
+statespace::StateSpacePtr
+  SE2BoxConstraintSampleGenerator::getStateSpace() const
+{
+  return mSpace;
+}
+
+//=============================================================================
+bool SE2BoxConstraintSampleGenerator::sample(
+  statespace::StateSpace::State* _state)
+{
+  Eigen::Vector3d tangent;
+
+  for (auto i = 0; i < tangent.size(); ++i)
+    tangent[i] = mDistributions[i](*mRng);
+
+  mSpace->expMap(tangent, static_cast<statespace::SE2::State*>(_state));
+
+  return true;
+}
+
+//=============================================================================
+int SE2BoxConstraintSampleGenerator::getNumSamples() const
+{
+  return NO_LIMIT;
+}
+
+//=============================================================================
+bool SE2BoxConstraintSampleGenerator::canSample() const
+{
+  return true;
+}
+
+//=============================================================================
+SE2BoxConstraint
+  ::SE2BoxConstraint(
+      std::shared_ptr<statespace::SE2> _space,
+      std::unique_ptr<util::RNG> _rng,
+      const Eigen::Vector2d& _lowerLimits,
+      const Eigen::Vector2d& _upperLimits)
+  : mSpace(std::move(_space))
+  , mRng(std::move(_rng))
+  , mRnDimension(2)
+  , mDimension(3)
+{
+  mLowerLimits[0] = -M_PI;
```

Comment:
:+1: to adding documentation for this.

Also, I wonder why we don't take the bounds for the rotation angle of SE(2) here. @gilwoolee Do we have a reason not to do so?